sources/src/test/java/com/google/solutions/jitaccess/core/catalog/project/TestProjectRoleActivator.java

```diff
@@ -58,32 +64,37 @@ public class TestProjectRoleActivator {
   public void provisionAccessForJitRequest() throws Exception {
     var resourceManagerClient = Mockito.mock(ResourceManagerClient.class);
     var activator = new ProjectRoleActivator(
-      Mockito.mock(EntitlementCatalog.class),
-      resourceManagerClient,
-      Mockito.mock(JustificationPolicy.class));
+        Mockito.mock(RequesterPrivilegeCatalog.class),
+        resourceManagerClient,
+        Mockito.mock(JustificationPolicy.class));
 
-    var request = activator.createJitRequest(
-      SAMPLE_REQUESTING_USER,
-      Set.of(
+    var privilege = new RequesterPrivilege<ProjectRoleBinding>(
         new ProjectRoleBinding(new RoleBinding(SAMPLE_PROJECT, SAMPLE_ROLE_1)),
-        new ProjectRoleBinding(new RoleBinding(SAMPLE_PROJECT, SAMPLE_ROLE_2))),
-      "justification",
-      Instant.now(),
-      Duration.ofMinutes(5));
+        SAMPLE_ROLE_1,
+        new SelfApproval(),
+        Status.AVAILABLE);
 
-    var activation = activator.activate(request);
+    var request = activator.createActivationRequest(
+        SAMPLE_REQUESTING_USER,
+        Set.of(),
+        privilege,
+        "justification",
+        Instant.now(),
+        Duration.ofMinutes(5));
+
+    var activation = activator.approve(SAMPLE_REQUESTING_USER, request);
 
     assertNotNull(activation);
     assertSame(request, activation.request());
 
-    verify(resourceManagerClient, times(2))
-      .addProjectIamBinding(
-        eq(SAMPLE_PROJECT),
-        argThat(b ->
-            IamTemporaryAccessConditions.isTemporaryAccessCondition(b.getCondition().getExpression()) &&
-            b.getCondition().getTitle().equals(JitConstraints.ACTIVATION_CONDITION_TITLE)),
-        eq(EnumSet.of(ResourceManagerClient.IamBindingOptions.PURGE_EXISTING_TEMPORARY_BINDINGS)),
-        eq("Self-approved, justification: justification"));
+    verify(resourceManagerClient, times(1))
```

Comment:
Why this change from 2 to 1? What does it mean in this context?

Reply:
Before we were testing that when we added IAM bindings for an `ActivationRequest` with two associated roles, then two IAM bindings were added. Since now we can only have one associated role also only one IAM binding should be added.